packages/next/next-server/server/next-server.ts

```diff
@@ -324,10 +325,32 @@ export default class Server {
 
       let defaultLocale = i18n.defaultLocale
       let detectedLocale = detectLocaleCookie(req, i18n.locales)
-      let acceptPreferredLocale =
-        i18n.localeDetection !== false
-          ? accept.language(req.headers['accept-language'], i18n.locales)
-          : detectedLocale
+      let languages = new Negotiator(req).languages()
+      let acceptPreferredLocale = detectedLocale
+
+      if (i18n.localeDetection !== false) {
+        try {
+          acceptPreferredLocale = ResolveLocale(
+            new Set(i18n.locales),
+            languages,
+            {
+              localeMatcher: 'best fit',
+            },
+            [],
+            {},
+            () => defaultLocale
+          ).locale
+        } catch (e) {
```

Comment:
Not sure about this part. Looks like you forgot the cookie detection when `i18n.localeDetection` is false.

If I were you, I would stick to the previous if / else statement to make sure everything stays the same (as I did in my PR).

Reply:
nah logic's the same